ci-operator/step-registry/telco5g/origin/tests/telco5g-origin-tests-commands.sh

```diff
@@ -44,30 +41,236 @@ git clone -b "${CNF_BRANCH}" "${CNF_REPO}" cnf-features-deploy
+  hosts: hypervisor
+  gather_facts: false
+  any_errors_fatal: true
+
+  tasks:
+  - name: Copy OpenShift's conformance test suite through bastion host to installer, run it and retrieve results
+    block:
+    - name: Create script to run openshift-tests binary on bastion host
+      ansible.builtin.copy:
+        content: |
+          #!/bin/bash
+
+          set -o nounset
+          set -o errexit
+          set -o pipefail
+
+          cd /home/cloud-user/
+
+          export KUBECONFIG=kubeconfig
+
+          ./openshift-tests run "${TEST_SUITE}" \\
+            --provider '${TEST_PROVIDER}' \\
+            -o e2e.log --junit-dir junit --file tests.txt \\
+            > openshift-tests.log
+        dest: '/tmp/kcli_${CLUSTER_NAME}_run-openshift-tests.sh'
+        mode: 0755
+
+    - name: Copy openshift-tests binary to bastion host
+      ansible.builtin.copy:
+        src: '${PWD}/openshift-tests'
+        dest: '/tmp/kcli_${CLUSTER_NAME}_openshift-tests'
+
+    - name: Copy list of tests to bastion host
+      ansible.builtin.copy:
+        src: '${ARTIFACT_DIR}/${TEST_FILE}'
+        dest: '/tmp/kcli_${CLUSTER_NAME}_${TEST_FILE}'
+
+    - name: Copy script to run openshift-tests binary to installer
+      ansible.builtin.shell:
+        cmd: |
+          kcli scp '/tmp/kcli_${CLUSTER_NAME}_run-openshift-tests.sh' \\
+            'cloud-user@${CLUSTER_NAME}-installer:/home/cloud-user/run-openshift-tests.sh' \\
+          && kcli ssh 'cloud-user@${CLUSTER_NAME}-installer' 'chmod a+x /home/cloud-user/run-openshift-tests.sh'
+
+    - name: Copy openshift-tests binary to installer
+      ansible.builtin.shell:
+        cmd: |
+          kcli scp '/tmp/kcli_${CLUSTER_NAME}_openshift-tests' \\
+            'cloud-user@${CLUSTER_NAME}-installer:/home/cloud-user/openshift-tests' \\
+          && kcli ssh 'cloud-user@${CLUSTER_NAME}-installer' 'chmod a+x /home/cloud-user/openshift-tests'
+
+    - name: Copy list of tests to installer
+      ansible.builtin.shell:
+        cmd: |
+          kcli scp '/tmp/kcli_${CLUSTER_NAME}_${TEST_FILE}' \\
+            'cloud-user@${CLUSTER_NAME}-installer:/home/cloud-user/tests.txt' \\
+
+    - name: Fix permissions for kubectl on installer
+      ansible.builtin.shell:
+        cmd: kcli ssh 'root@${CLUSTER_NAME}-installer' 'chmod a+x /usr/bin/kubectl'
+
+    - name: Make kubeconfig available to non-root user on installer
+      ansible.builtin.shell:
+        cmd: |
+          kcli ssh 'root@${CLUSTER_NAME}-installer' "sh -c '\\
+              cp /root/ocp/auth/kubeconfig /home/cloud-user/kubeconfig \\
+              && chown cloud-user /home/cloud-user/kubeconfig \\
+          '"
+
+    - name: Test kubeconfig for non-root user on installer
+      ansible.builtin.shell:
+        cmd: |
+          kcli ssh 'cloud-user@${CLUSTER_NAME}-installer' "sh -c 'export KUBECONFIG=kubeconfig && oc get nodes'"
+
+    - name: Run script for openshift-tests binary on installer
```

Comment:
Oh, I see, so you run it as non-root user on installer. Is there a problem with running as root? I think it can reduce amount of tasks. Installer machine is actually supposed to be used once, that's why it has all-root there.
If it's not a problem to run as root from openshift-tests perspective, it's much preferable

Reply:
For security reasons I always run code unprivileged but if you prefer root we can do that o/c.